### DISCUSSION TITLE
`.ics` File Editing in ICSx⁵

### DISCUSSION THREAD

**Discussion by bgodlin** [bitfireAT/icsx5#168] [Ideas]
Hello,

I'm a devoted user of ICSx⁵ and want to start by commending the seamless calendar synchronization it offers. Your efforts to simplify this process are truly appreciated.

Currently, I use ICSx⁵ on my mobile alongside Mozilla Thunderbird on my PC. This combination has been transformative. However, I've encountered a minor hurdle. 

With the growing popularity of P2P software like Syncthing, the need to address this issue becomes more pronounced. While I can create events on Thunderbird that seamlessly sync to my mobile, the reverse isn't as possible due to the lack of a solution that enables local .`ics` files editing for subsequent syncing to the PC.

Therefore, I propose a straightforward enhancement: introducing a `.ics` file editing feature in ICSx⁵, similar to what's available in Thunderbird (as outlined here: https://wiki.mozilla.org/Calendar:FAQ#How_can_I_subscribe_to_shared_iCalendar_file). This upgrade would empower users to edit or create events on mobile devices, with the assurance that these changes would synchronize seamlessly using tools like Syncthing, eliminating the need for centralized server-hosted calendars.

For further context, I direct your attention to the ongoing discussion within the repository: https://github.com/bitfireAT/icsx5/discussions/90. This thread encapsulates the essence of my suggestion.

Thank you for considering this idea. Your ongoing commitment to improvement is highly valued.

**Comment by ArnyminerZ** [comment]
The scope of this app is really simple, this is for reading subscriptions from somewhere, and importing them into your local calendar. Example use cases are schedules, events and this kind of things, it's not intended as a synchronization service for your own events but for events which you cannot edit.

.ics file importing is here because some providers only offer the possibility to download the ICS file, but not having a stable endpoint for downloading the updated events list.

We can discuss file editing, but I don't think it's something we desire, since we want to keep this app as simple as possible. @rfc2822 or @devvv4ever can further dig into this.

Whatsoever, I completely understand your use-case, and maybe it's something we should consider.

**Reply by rfc2822** [reply]
I agree: currently not in scope, but maybe will be considered in the future (but I wouldn't count on it).

https://github.com/bitfireAT/icsx5/discussions/90#discussioncomment-6779096

**Comment by adapt-L** [comment]
This would be a cool feature I think. Would it be in scope for davx5?

**Reply by ArnyminerZ** [reply]
DAVx⁵'s scope is to use the DAV protocol, so adding ICS file editing would be a bit out of what DAVx⁵ is supposed to do. However, nothing is completely undoable.

Anyway, I think that in any case this feature should be in ICSx⁵, and not DAVx⁵. But again, not in any time soon, other more important things are in the way for both apps.
